### PR TITLE
Reference Scoping Fix

### PR DESCRIPTION
**Pull request by tcormackMW**
with to references to prototype services. Also added test cases that validate proper behavior for references to both singleton and prototype services.

### CODE REVIEW

**Comment by tcormackMW** [comment]
Verify what we do for bundle scoped (whether this defaults to singleton or prototype

File: compendium/DeclarativeServices/src/ComponentContextImpl.cpp [hunk 1]
```diff
@@ -69,50 +69,56 @@ namespace cppmicroservices
...
+                                      else
+                                      {
+                                          cppmicroservices::ServiceObjects<void> sObjs = bc.GetServiceObjects(sRefU);
+                                          auto interfaceMap = sObjs.GetService();
+                                          if (interfaceMap)
+                                          {
+                                              foundAtLeastOneValidBoundService = true;
+                                              serviceMap.push_back(interfaceMap);
+                                          }
+                                      }
```
Comment (jeffdiclemente) [inline compendium/DeclarativeServices/src/ComponentContextImpl.cpp:107]: I think this entire block can be simplified significantly and that the reference scope of the service reference doesn't need to be queried at all based off of the documentation for `ServiceObjects` class. `ServiceObjects` should do the right thing already based on the scope of the service.

**Review by jeffdiclemente — CHANGES_REQUESTED** [review]

File: compendium/test_bundles/TestBundleDSFAC1/src/ServiceImpl.hpp [hunk 2]
```diff
@@ -22,14 +22,129 @@ namespace sample
         ServiceAImpl(ServiceAImpl&&) = delete;
         ServiceAImpl& operator=(ServiceAImpl const&) = delete;
         ServiceAImpl& operator=(ServiceAImpl&&) = delete;
- 
+
         ~ServiceAImpl() = default;
 
+        void*
```
[the remark this answers is not in the document]
Comment (carneyweb) [reply compendium/test_bundles/TestBundleDSFAC1/src/ServiceImpl.hpp:28]: Done.

File: compendium/test_bundles/TestBundleDSFAC1/src/ServiceImpl.hpp [hunk 3]
```diff
@@ -22,14 +22,129 @@
...
+        {
+        }
+
+        cppmicroservices::AnyMap
+        GetProperties() override
+        {
+            return {};
+        }
+
+        void*
```
[the remark this answers is not in the document]
Comment (carneyweb) [reply compendium/test_bundles/TestBundleDSFAC1/src/ServiceImpl.hpp:63]: Done.

File: compendium/test_bundles/TestBundleDSFAC1/src/ServiceImpl.hpp [hunk 4]
```diff
@@ -22,14 +22,129 @@
...
+        Modified(std::shared_ptr<ComponentContext> const&, std::shared_ptr<cppmicroservices::AnyMap> const&)
+        {
+        }
+        cppmicroservices::AnyMap
+        GetProperties() override
+        {
+            return {};
+        }
+
+        void*
```
[the remark this answers is not in the document]
Comment (carneyweb) [reply compendium/test_bundles/TestBundleDSFAC1/src/ServiceImpl.hpp:94]: Done.

File: compendium/DeclarativeServices/test/gtest/TestComponentContextImpl.cpp [hunk 5]
```diff
@@ -157,6 +154,7 @@ namespace cppmicroservices
             });
         }
 
+#if NEVER
```
[the remark this answers is not in the document]
Comment (carneyweb) [reply compendium/DeclarativeServices/test/gtest/TestComponentContextImpl.cpp:157]: Done.

**Review by jeffdiclemente — APPROVED** [review]

**Comment by tcormackMW** [comment]
> Verify what we do for bundle scoped (whether this defaults to singleton or prototype)

@jeffdiclemente It looks like this doesn't matter anymore because we rely on GetService to return the 'correct' thing based on the scope of the service. 

For the sake of figuring this question out anyway, we treat BUNDLE_SCOPE as PROTOTYPE_SCOPE in the codebase.

**Pull request merged**